terraform/modules/PersistGrantsGovXMLDB/main.tf

```diff
@@ -0,0 +1,117 @@
+terraform {
+  required_version = "1.3.9"
+  required_providers {
+    aws = "~> 4.55.0"
+  }
+}
+
+locals {
+  dd_tags = merge(
+    {
+      for item in compact(split(",", try(var.additional_environment_variables.DD_TAGS, ""))) :
+      split(":", trimspace(item))[0] => try(split(":", trimspace(item))[1], "")
+    },
+    var.datadog_custom_tags,
+    { handlername = lower(var.function_name), },
+  )
+}
+
+data "aws_s3_bucket" "prepared_data" {
+  bucket = var.grants_prepared_data_bucket_name
+}
+
+module "lambda_execution_policy" {
+  source  = "cloudposse/iam-policy/aws"
+  version = "0.4.0"
+
+  iam_source_policy_documents = var.additional_lambda_execution_policy_documents
+  iam_policy_statements = {
+    AllowInspectS3PreparedData = {
+      effect = "Allow"
+      actions = [
+        "s3:GetObject",
+        "s3:ListBucket"
+      ]
+      resources = [
+        data.aws_s3_bucket.prepared_data.arn,
+        "${data.aws_s3_bucket.prepared_data.arn}/*/*/grants.gov/v2.xml"
+      ]
+    }
+    AllowDynamoDBPreparedData = {
+      effect = "Allow"
+      actions = [
+        "dynamodb:ListTables",
+        "dynamodb:BatchGetItem",
+        "dynamodb:GetItem",
+        "dynamodb:Query",
+        "dynamodb:Scan",
+        "dynamodb:BatchWriteItem",
+        "dynamodb:PutItem",
+        "dynamodb:UpdateItem"
+      ]
```

Comment:
I'm not sure all of these permissions are necessary (please feel free to push back if you've seen an indication that they are!).

I think the following should be sufficient (**note:** the two I'm not positive about are `dynamodb:GetItem` and `dynamodb:PutItem` – I don't _think_ they're required as underlying support for `UpdateItem` operations, but I could be wrong about that):
```suggestion
      actions = [
        "dynamodb:ListTables",
        "dynamodb:UpdateItem"
      ]
```

Reply:
I just tested the lambda with these reduced permissions and it worked without issue.  PR updated.